mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1250,37 +1314,52 @@ public bool ConnectAsync (SocketAsyncEventArgs e)
 			if (e.RemoteEndPoint == null)
 				throw new ArgumentNullException ("remoteEP");
 
-			e.curSocket = this;
-			e.Worker.Init (this, e, SocketOperation.Connect);
+			SocketAsyncResult sockares = new SocketAsyncResult (this, s => SocketAsyncEventArgsCallback (ConnectAsyncCallback, s), e, SocketOperation.Connect);
```

Comment:
In all the other methods, `sockares` is being used over `result`, so using `result` makes it hard to see what it represents.